command/git.go

```diff
@@ -193,9 +195,16 @@ func (c *gitFetchProject) ParseParams(params map[string]interface{}) error {
 	return nil
 }
 
+func (c *gitFetchProject) gitResetWithUnshallow(revision string) []string {
+	return []string{
+		fmt.Sprintf("if $(git reset --hard %s); then exit 0; fi", revision),
+		fmt.Sprintf("if $(git fetch --help | grep -q unshallow); then git fetch --unshallow; else git fetch --depth %d", math.MaxInt32),
```

Comment:
maybe an alternative is to try one and if it fails try the second?

Reply:
I'd like to avoid that because it makes the behavior non-deterministic. Checking the man page means that Evergreen will always do the same thing. Since `git fetch --unshallow` could fail for reasons other than unshallow not being present as an option, sometimes Evergreen would `git fetch --depth`.

Reply:
I suppose an alternative would be to always use `git fetch --depth` or to allow us to selectively opt in to this for distros.